homeassistant/components/casatunes/media_player.py

```diff
@@ -47,41 +45,32 @@
     """Set up the CasaTunes platform."""
     from casatunes import CasaTunes
 
-    host = config.get(CONF_HOST)
-    port = config.get(CONF_PORT)
+    host = config[CONF_HOST]
+    port = config[CONF_PORT]
+    session = async_get_clientsession(hass)
 
-    ct = CasaTunes(host, port)
+    ct = CasaTunes(host, port, session)
 
     sources = await ct.get_sources()
     zones = await ct.get_zones()
 
-    zone_devices = []
-    player_devices = []
+    devices = []
 
     if sources is False:
         _LOGGER.error("Sources response is incorrect")
-        return False
+        raise PlatformNotReady
 
     if zones is False:
         _LOGGER.error("Zones response is incorrect")
-        return False
+        raise PlatformNotReady
 
     for zone in zones:
```

Comment:
In python you can use list comprehension: `devices.extend([CasaTunesZone(zone) for zone in zones])`